server/models/User.py

```diff
@@ -30,6 +47,17 @@ def save(self):
         db.session.add(self)
         db.session.commit()
 
+    def update(self, data):
+        for key, item in data.items():
+            if key == 'password':
+                self.password = self.__generate_hash(item)
```

Comment:
Does this make this happen twice? Since you do it in __init__ and in update as well? How do you know self.password is not already hashed? I'd want this to happen in one spot so we are not likely to make a mistake

Reply:
You could override the setter for this class... that would be really clean but may be a bit involved in terms of Python OOP. The idea would be that whenever you do self.password =, it will automatically hash the password. This is something to just know that you could do, however, I'd actually not recommend it, since it makes the code less readable.

Reply:
@skassam21 interesting idea about overriding the setter to hash automatically. its a neat trick.  i will keep hashing code as it currently is though.  password gets hashed on user creation. it also gets hashed if user makes a put request with password in the request body